src/mudlet.cpp

```diff
@@ -2341,16 +2352,15 @@ void mudlet::slot_notes()
 
 void mudlet::slot_irc()
 {
-    if( ! mpIRC )
+    Host * pHost = getActiveHost();
+    if( ! pHost ) return;
+
+    if( ! mpIrcClientMap[pHost] )
     {
-        mpIRC = new dlgIRC();
-        mpIRC->setWindowTitle( tr( "Mudlet live IRC Help Channel #mudlet-help on irc.freenode.net" ) );
-        mpIRC->setWindowIcon( QIcon( QStringLiteral( ":/icons/mudlet_irc.png" ) ) );
```

Comment:
Icon and title were moved to the constructor of `dlgIRC` I think I have already added the host profile name but adding the server name and nick is a good idea.